restock-shop.lic

```diff
@@ -12,98 +12,95 @@ class ShopRestock
-      ['cutlass', 120_000, 7, 'on rack', 'cutlass', 'steel'],
-      ['bola', 120_000, 5, 'on rack', 'bola', 'steel'],
-      ['hammer', 120_000, 10, 'on rack', 'throwing hammer', 'steel'],
-      ['akabo', 120_000, 13, 'on rack', 'akabo', 'steel'],
-      ['spike', 100_000, 4, 'on rack', 'throwing spike', 'steel'],
-      ['nightstick', 120_000, 7, 'on rack', 'nightstick', 'steel'],
-      ['greataxe', 120_000, 12, 'on rack', 'greataxe', 'steel'],
-      ['carryall', 200_000, 5, 'on table', 3, 'carryall', 'heavy silk'],
-      ['haversack', 200_000, 6, 'on table', 3, 'haversack', 'heavy silk']
-    ]
-
-    walk_to(@shop_id)
-    fput('open entrance')
-    pause 0.5
-    fput('go entrance')
-
+    @settings = get_settings
+
+    @shop_id = @settings.restock_shop_outside_room
+    @inner_id = @settings.restock_shop_inside_room
+    @entrance = @settings.restock_shop_entrance_noun
+    @item_list = @settings.restock_shop_items
+    @town = @settings.restock_shop_town
+    @part_data = get_data('crafting')[:recipe_parts]
+    @bag = @settings.crafting_container
+    # TODO: Validate users item list against recipe data. Ensure that there is only one match per user specified recipe.
+    # TODO: Validate users town. Allow only towns with plazas for simplicity
+
+    # @item_list = [
+    #   ['backpack', 100_000, 5, 'on table', 3, 'backpack', 'heavy linen'],
+    #   ['bag', 100_000, 6, 'on table', 3, 'duffel bag', 'heavy linen'],
+    #   ['sealed', 220_000, 8, 'in cabinet', 'gryphon'],
+    #   ['sword', 140_000, 9, 'on rack', 'bastard sword', 'steel'],
+    #   ['mace', 140_000, 12, 'on rack', 'bar mace', 'steel'],
+    #   ['hammer', 120_000, 10, 'on rack', 'throwing hammer', 'steel'],
+    #   ['greataxe', 120_000, 12, 'on rack', 'greataxe', 'steel'],
+    #   ['carryall', 200_000, 5, 'on table', 3, 'carryall', 'heavy silk'],
+    #   ['haversack', 200_000, 6, 'on table', 3, 'haversack', 'heavy silk']
+    # ]
+    go_to_shop
     check_surfaces
 
-    return if Room.current.id == @inner_id
-    walk_to(@shop_id)
-    fput('open entrance')
-    pause 0.5
-    fput('go entrance')
+    go_to_shop
   end
 
   def check_surfaces
     missing_items = []
-    @item_list.map { |x| x[3] }.uniq.each do |surface|
+    @item_list.map { |data| data['surface'] }.uniq.each do |surface|
       raw_list = bput("look #{surface}", "There's nothing", 'you see [^\.]*')
       raw_list = if raw_list == "There's nothing"
                    []
                  else
-                   list_to_nouns(raw_list)
+                   list_to_array(raw_list)
                  end
-      missing_items += @item_list.select { |data| surface == data[3] }.reject { |data| raw_list.include?(data.first) }
+      missing_items += @item_list.select { |item| surface == item['surface'] }
+                                 .reject { |item| raw_list.find { |raw_item| raw_item.match(item['full_name']) } }
     end
+
+    return if missing_items.empty?
+
+    respond 'Restocking'
```

Comment:
💡 What's the difference between `respond` and `echo`?

Reply:
respond is a 'cleaner' output in my opinion, it doesn't prepend the script name to the output. So you will see output as 
```
Restocking
----------------
- quilted heavy burlap gloves
- quilted heavy burlap hood
- plate mask
```

Reply:
Thanks 👍